### PR TITLE
Add Dockerfile for building image for Docker Hub

### PR DESCRIPTION
**Pull request by jwodder**
Needed for https://github.com/dandi/dandi-cli/issues/164

### CODE REVIEW

File: dev/django-public.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FROM python:3.8
```
Comment (yarikoptic) [inline dev/django-public.Dockerfile:1]: Would `-slim` (165M) or `-alpine` (43M) be enough?  python:3.8 is quite heavier (800M?)

Comment (jwodder) [reply dev/django-public.Dockerfile]: `-slim` works.

**Comment by yarikoptic** [comment]
Dear @dandi/dandiarchive people, we need your blessing or cursing (blessing would be more appreciated, merge most appreciated, and cursing we can tollerate/workaround ;)) to proceed forward!

**Comment by yarikoptic** [comment]
Sorry for rushing it, but since AFAIK it should not break anything, and would let us go forward -- I will just merge

**Pull request merged**